pkg/util/docker/docker_test.go

```diff
@@ -91,7 +92,139 @@ func TestResolveImageName(t *testing.T) {
+				ContainerJSONBase: &types.ContainerJSONBase{Image: imageWithShaTag},
+				Config:            &container.Config{Image: "myapp"},
+			},
+			expectedImage: imageName,
+		}, {
+			name: "test image sha tag itself",
+			input: types.ContainerJSON{
+				ContainerJSONBase: &types.ContainerJSONBase{Image: imageSha},
+				Config:            &container.Config{Image: "myapp"},
+			},
+			expectedImage: imageName,
+		},
+	} {
+		result, err := globalDockerUtil.ResolveImageNameFromContainer(tc.input)
+		assert.Equal(tc.expectedImage, result, "%s test failed; expected %s but got %s", tc.name, tc.expectedImage, result)
+		assert.Nil(err, "%s test failed; expected nil error but got %s", tc.name, err)
+	}
+}
+
+func TestResolveImageNameFromContainerError(t *testing.T) {
+	imageSha := "sha256:bdc7dc8ba08c2ac8c8e03550d8ebf3297a669a3f03e36c377b9515f08c1b4ef4"
+	assert := assert.New(t)
+
+	// This returns a nil client because the transport verification fails
+	cli, _ := client.NewEnvClient()
+
+	globalDockerUtil = &DockerUtil{
+		cfg:            &Config{CollectNetwork: false},
+		cli:            cli,
+		imageNameBySha: make(map[string]string),
+	}
+
+	input := types.ContainerJSON{
+		ContainerJSONBase: &types.ContainerJSONBase{Image: imageSha},
+		Config:            &container.Config{Image: "myapp"},
+	}
+
+	result, err := globalDockerUtil.ResolveImageNameFromContainer(input)
+	assert.Equal(imageSha, result, "test failed; expected %s but got %s", imageSha, result)
+	assert.NotNil(err, "test failed; expected an error but got %s", err)
+}
 
+func TestGetBestImageName(t *testing.T) {
+	latest := "latest"
+	assert := assert.New(t)
+	for _, tc := range []struct {
+		name          string
+		imageInspect  types.ImageInspect
+		configImage   string
+		expectedImage string
+	}{
+		{
+			name: "only one repo tag",
+			imageInspect: types.ImageInspect{
+				ID:       "image_id",
+				RepoTags: []string{latest},
+			},
+			configImage:   latest,
+			expectedImage: latest,
+		}, {
+			name: "two repo tags and configImage matches one",
+			imageInspect: types.ImageInspect{
+				ID:       "image_id",
+				RepoTags: []string{latest, "random_tag"},
+			},
+			configImage:   latest,
+			expectedImage: latest,
+		}, {
+			name: "two repo tags and configImage does not match one",
+			imageInspect: types.ImageInspect{
+				ID:       "image_id",
+				RepoTags: []string{"random_tag1", "random_tag2"},
+			},
+			configImage:   latest,
+			expectedImage: "random_tag1",
+		}, {
+			name: "no repo tags but a repo digest",
+			imageInspect: types.ImageInspect{
+				ID:          "image_id",
+				RepoDigests: []string{"quay.io/foo/bar@sha256:hash"},
+			},
+			configImage:   latest,
+			expectedImage: "quay.io/foo/bar",
+		}, {
+			name: "no repo tags or repo digests, returns empty image name",
+			imageInspect: types.ImageInspect{
+				ID: "image_id",
+			},
+			configImage:   latest,
+			expectedImage: "",
+		},
+	} {
+		result := getBestImageName(tc.imageInspect, tc.configImage)
```

Comment:
Nit - might want to wrap these two lines in `t.Run(...`

Reply:
Hm, I mimicked the format of the other tests in the file; think it's worth updating all of them?

Reply:
as discussed offline i'll make a separate follow up PR to change all of the tests to do this